### PR TITLE
#159053987 Add functionality for rating articles

### PR DESCRIPTION
**Pull request by davenjeru**
**What does this PR do?**
Adds the functionality to view an article's ratings.and to rate an article.

**Description of Task to be completed?**
1. Add a new app called Ratings with its model that reference both the User and Article model
2. Modify ArticleSerializer to have two new fields, **ratingsCount** and **averageRating**.
3. Add a method to enable a user to rate an article with a number of stars.

**How can this be manually tested?**
- Clone the repository from [here](https://github.com/andela/ah-scorpion/tree/ft-Article-Rating-%23159053987)
- Activate your virtual environment and use the `.env-sample` to set the environment variables.
- Use postman to test these endpoints.
  `GET api/v1/articles/<str:slug>/`, the response should look something like this(notice the two new fields):
```source-json
{
  "article": {
    "slug": "how-to-train-your-dragon",
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian",
    "tagList": ["dragons", "training"],
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "favorited": false,
    "favoritesCount": 0,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    },
    "averageRating": null,
    "ratingsCount": 0,
  }
}
``` 

`GET api/v1/articles/<str:slug>/ratings` the response should be something like this(or an empty list if the ratingsCount is 0):
```source-json
{
  "ratings": [
    {
      "user": "jake",
      "stars": 5
    },
    {
      "user": "jamie",
      "stars": 3
    }
  ]
}
```

`POST api/v1/articles/<str:slug>/ratings` with the following request body in JSON:
```source-json
{
  "stars": 4
}
```

The response should be something like this:
```source-json
{
  "ratings": {
      "user": "jake",
      "stars": 4
    }
}
```

**What are the relevant pivotal tracker stories?**
[#159053987](https://www.pivotaltracker.com/story/show/159053987)

### CODE REVIEW

**Review by munyili-samuel — APPROVED** [review]
I have tested the PR and it works as shown on the description.
The tests are also passing.
Good job

**Comment by kipropbrian** [comment]
Hey Dave, the title of this PR does not follow andela guidelines.

**Review by hoslack — APPROVED** [review]
Great work there

**Comment by dmigwi** [comment]
Your commit messages are too repetitive. They should be concise and descriptive without any repeated words wherever possible.

screen shot 2018-08-12 at 02 39 06

You will need to rebase your branch to either squash all commits into one commit or Edit all the commit messages.

File: authors/apps/ratings/tests.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+class RatingsTestCase(APITestCase):
+    def setUp(self):
+        self.reader = self.create_user()
+        self.author = self.create_user(
+            email='author@email.com',
+            username='author',
+        )
+        self.slug = self.create_article().data['slug']
+        self.client = APIClient()
+        pass
```
Comment (RuthNjeri) [inline authors/apps/ratings/tests.py]: I do not think `pass` is necessary here.

**Review by RuthNjeri — CHANGES_REQUESTED** [review]

File: authors/apps/ratings/serializers.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        """
+        When the URL is accessed with the POST method this function is hit.
+
+        :param validated_data:
+        :return:
+        """
+        # set the article given from the context. It was set in our view
+        article = self.context['article']
+
+        # If the article is not found the raise error
```
Comment (lennykioko) [inline authors/apps/ratings/serializers.py]: I think you meant to say **then** and not **the**

File: authors/apps/ratings/urls.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from django.urls import path
+
+from . import views
+
+app_name = "articles"
```
Comment (lennykioko) [inline authors/apps/ratings/urls.py]: I think that the `app_name` here should be ratings since we are currently in the ratings app.

**Comment by RuthNjeri** [comment]
The tests work well and the requests return the correct payload.  Good job.

**Review by lennykioko — APPROVED** [review]

**Review by RuthNjeri — APPROVED** [review]

**Pull request merged**